client/lib/analytics/signup.js

```diff
@@ -66,7 +67,8 @@ export function recordSignupComplete(
 
 	if ( isNew7DUserSite ) {
 		// Tracks
-		recordTracksEvent( 'calypso_new_user_site_creation', { flow } );
+		const device = resolveDeviceTypeByViewPort();
+		recordTracksEvent( 'calypso_new_user_site_creation', { flow, device } );
```

Comment:
Nitpick: This is technically not device but viewport. I could still be on desktop and yet get the value "tablet".

Reply:
Yes @niranjan-uma-shankar this is a convention to match the future function which is resolveDeviceByUserAgent.